README.md

```diff
@@ -367,3 +367,31 @@ Promise.resolve("Hello!")
 ```
 
 If you don't `catch` a rejected promise, it will throw an error and crash (the same as if you didn't catch an exception in a try/catch statement).
+
+# Persisting the state
+
+Let's make sure that changes in our list are persisted between apps restarts. To do that, we will be using [`AsyncStorage`](https://facebook.github.io/react-native/docs/asyncstorage.html).
+
+To retrieve something stored, we can use `getItem`: it accepts a key and it returns a promise, the result will be either the previously stored data, or `null` if there isn't any.
+
+To retrieve the state when the application is initialized, we can use one of the lifecycle callbacks of a component (they're present only in class components), [`componentDidMount`](https://reactjs.org/docs/react-component.html#componentdidmount) and set the retrieved state inside the promise callback.
+
+```javascript
+componentDidMount() {
```

Comment:
is it recommend for each component to query its initial state or should it be passed in from a parent/page level component? 

If passed from a parent we can set the initial state in the constructor from the props and avoid the initial double render :ghost:  


Is it worth warning about modifying the state in `componentDidMount` will cause a double initial render? Or can this be inferred by all state changes cause a render? (assuming the shouldComponentUpdate passes)

Reply:
> is it recommend for each component to query its initial state or should it be passed in from a parent/page level component?

imho (this is an opinionated doc, after all, since I'm the only author 👻) it is, cause children components already have a responsibility (rendering the data)

> If passed from a parent we can set the initial state in the constructor from the props and avoid the initial double render 👻

this is possible only if you can retrieve the state synchronously, which we can't if reading from disk

> Is it worth warning about modifying the state in componentDidMount will cause a double initial render?

I wouldn't bother at this stage, it's mostly ok (perf issue but at this scale it isn't really an issue)

> Or can this be inferred by all state changes cause a render? (assuming the shouldComponentUpdate passes)

Not all state changes cause a render 👅 state changes are async, and are batched

Reply:
I was pushing towards initial states and then updates later on but allow the lower level components to not have to worry about it. 

```jsx
class PageComponent extends Component {

  state = {
    todos: { values: [] }
  }

  render = () => (<Todos {...state.todos}/>)

  componentDidMount = () => {
    AsyncStorage.get('todos').then(todos => this.setState({todos: todos}))  
  }

}

const Todos = ( {values} ) => (
  ...
)

```

Your points are all relevant still! I'm just getting carried away :joy:

Reply:
@ouchadam any strong feelings
@fourlastor working on changes?

otherwise :shipit:

Reply:
Yep we chatted offline: basically we were agreeing on the same thing 😄